src/com/fundation/search/controller/Validator.java

```diff
@@ -0,0 +1,108 @@
+package com.fundation.search.controller;
+
+/*
+ * @(#)FileClass.java
+ *
+ * Copyright (c) 2018 Jala Foundation.
+ * Address
+ * All rights reserved.
+ *
+ * This software is the confidential and proprietary information of
+ * Jala Foundation, ("Confidential Information").  You shall not
+ * disclose such Confidential Information and shall use it only in
+ * accordance with the terms of the license agreement you entered into
+ * with Sun.
+ */
+
+import java.io.File;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This class validate all the fields that are recivied by the GUI.
+ *
+ * @author Ariel Gonzales Vargas - AT-[06].
+ * @version 1.0.
+ */
+public class Validator {
+
+    /**
+     * Init the Validator.
+     */
+    public Validator() {
+    }
+
+    /**
+     * It method checks if the input is a valid Path format.
+     *
+     * @param path Path
+     * @return boolean
+     */
+    public boolean validatorPath(String path) {
+        Pattern pat = Pattern.compile("(^[A-Z]:(\\\\)*)?((\\\\)\\w+((\\s|\\.)\\w+)*)+");
+        Matcher mat = pat.matcher(path);
+        return mat.matches();
+
+    }
+
+    /**
+     * It method checks if the path exist.
+     *
+     * @param path Direction path.
+     * @return boolean.
+     */
+    public boolean isValidPath(String path) {
+        File pathD = new File(path);
+        return pathD.isDirectory();
+    }
+
+    /**
+     * It method checks if the input is a valid File format.
+     *
+     * @param file string
+     * @return boolean
+     */
+    public boolean validatorFile(String file) {
+        Pattern pat = Pattern.compile("(\\w+(\\s|[^:*?\"<>|]\\w+)*)+");
```

Comment:
You need declare "private Pattern pat" and "private Matcher mat", before your constructor